src/API.jl

```diff
@@ -904,25 +906,27 @@ function _is_stale(paths::Vector{String}, sourcepath::String, toml_c::Base.TOMLC
     return true
 end
 
-function _auto_precompile()
-    if parse(Int, get(ENV, "JULIA_PKG_PRECOMPILE_AUTO", "0")) == 1
-        Pkg.precompile(internal_call=true)
+function _auto_precompile(ctx::Context)
+    if parse(Int, get(ENV, "JULIA_PKG_PRECOMPILE_AUTO", "1")) == 1
+        Pkg.precompile(ctx, internal_call=true)
     end
 end
 
-precompile(;internal_call::Bool=false) = precompile(Context(), internal_call=internal_call)
-function precompile(ctx::Context; internal_call::Bool=false)    
+precompile(; kwargs...) = precompile(Context(); kwargs...)
+function precompile(ctx::Context; internal_call::Bool=false, io::IO=stderr)
     num_tasks = parse(Int, get(ENV, "JULIA_NUM_PRECOMPILE_TASKS", string(Sys.CPU_THREADS + 1)))
     parallel_limiter = Base.Semaphore(num_tasks)
-    
+    fancy_print = (io isa Base.TTY) && (get(ENV, "CI", nothing) != "true")
+
     # when manually called, unsuspend all packages that were suspended due to precomp errors
     !internal_call && Operations.precomp_unsuspend!() 
+    action_help = internal_call ? " (tip: to disable auto set ENV[\"JULIA_PKG_PRECOMPILE_AUTO\"]=0)" : ""
```

Comment:
I don't think we need this, will get pretty annoying the 100th time you see it.

Reply:
I think it is fine to have it for now, just until it has gotten a bit more usage. Then we remove it after it's been tested properly in the wild.

Reply:
I think you're right. I've removed it and added info about the automatic mode to the `Pkg.precompile` docs, including this tip on how to disable it

Reply:
Ok. I'll add it back.

Reply:
Sorry for the back and forth :D